src/main/kotlin/com/openlattice/search/renderers/CAREIssueAlertEmailRenderer.kt

```diff
@@ -0,0 +1,187 @@
+private val TITLE_FQN = FullQualifiedName("ol.title")
+private val COMPLETED_DATE_TIME_FQN = FullQualifiedName("date.completeddatetime")
+private val ENTRY_UPDATED_FQN = FullQualifiedName("general.entryupdated")
+
+
+/* METADATA TAGS */
+private const val PERSON_ENTITY_SET_ID_METADATA = "personEntitySetId"
+private const val STAFF_ENTITY_SET_ID_METADATA = "staffEntitySetId"
+private const val TIME_ZONE_METADATA = "timezone"
+private const val ASSIGNED_TO_ENTITY_SET_ID_METADATA = "assignedToEntitySetId"
+private const val REPORTED_ENTITY_SET_ID_METADATA = "reportedEntitySetId"
+
+/*
+*
+* Search metadata is expected to contain the following fields:
+* {
+*   "personEntitySetId": <UUID>,
+*   "staffEntitySetId": <UUID>,
+*   "assignedToEntitySetID": <UUID>,
+*   "reportedEntitySetID": <UUID>,
+*   "timezone": <TimeZones>,
+*   "alertName": <String>,
+* }
+*
+*/
+
+class CAREIssueAlertEmailRenderer {
+
+    companion object {
+
+        private fun isMatchByEntitySetId(neighbor: NeighborEntityDetails, entitySetId: UUID, associationEntitySetId: Optional<UUID>): Boolean {
+            val isMatchingEntitySetId = neighbor.neighborEntitySet.isPresent && neighbor.neighborEntitySet.get().id == entitySetId
+            return if (associationEntitySetId.isPresent) {
+                (neighbor.associationEntitySet.id == associationEntitySetId.get()) && isMatchingEntitySetId
+            }
+            else { isMatchingEntitySetId }
+        }
+
+        private fun getCombinedNeighbors(
+                neighbors: List<NeighborEntityDetails>, entitySetId: UUID, associationEntitySetId: Optional<UUID>
+        ): Map<FullQualifiedName, Set<Any>> {
+            val entity = mutableMapOf<FullQualifiedName, MutableSet<Any>>()
+            neighbors.filter { isMatchByEntitySetId(it, entitySetId, associationEntitySetId) }.forEach {
+                it.neighborDetails.get().forEach { (fqn, values) ->
+                    entity.getOrPut(fqn) { HashSet(values.size) }.addAll(values)
+                }
+            }
+
+            return entity
+        }
+
+        private fun getPersonDetails(
+                neighbors: List<NeighborEntityDetails>, personEntitySetId: UUID
+        ): HashMap<String, String> {
+            val combinedEntity = getCombinedNeighbors(neighbors, personEntitySetId, Optional.empty())
+
+            val tags = HashMap<String, String>(1)
+
+            val firstName = (combinedEntity[FIRST_NAME_FQN] ?: emptySet()).joinToString("/")
+            val lastName = (combinedEntity[LAST_NAME_FQN] ?: emptySet()).joinToString("/")
+            val middleName = (combinedEntity[MIDDLE_NAME_FQN] ?: emptySet()).joinToString("/")
+            tags["formattedName"] = "$lastName, $firstName $middleName"
+
+            return tags
+        }
+
+        private fun getReporterDetails(
+                neighbors: List<NeighborEntityDetails>, staffEntitySetId: UUID, reporterEntitySetId: UUID
+        ): HashMap<String, String> {
```

Comment:
Prefer interfaces to classes. Specifically return a Map<String, String> if it mutability is not required and MutableMap<String,String> if mutability is required.  Another thing (relatively minor),```mapOf( "reporter" to (combinedEntity[PERSON_ID_FQN] ?: emptySet()).joinToString(", ") ``` will create a special singleton map without having to resort to manually invoking the HashMap constructor.